sdk/cosmos/azure_data_cosmos/docs/control-plane-warning.md

```diff
@@ -0,0 +1,5 @@
+<div class="warning">
+
+This is a control-plane API and requires that you authenticate using a key. To use Entra ID to perform this operation, you must use the [Azure Resource Manager APIs](https://learn.microsoft.com/en-us/azure/templates/microsoft.documentdb/databaseaccounts).
```

Comment:
I think this PR does the right thing for the point in time we're at, but just pointing out that I believe there is a plan to support these operations via Entra ID at some point in 2025, and also to integrate the dataplane RBAC with normal Azure RBAC.

Reply:
Yep, the intent is that this feature can be "removed" later. It will have to remain present, and continue to imply `key_auth` to avoid breaking users, but the `#[cfg]` attributes referencing it would go away and the methods related to it would just become available to everyone. At that point users would be able to remove the feature reference from their Cargo.toml (possibly replacing with `key_auth` if they still need that support)